docs/error_codes.md

```diff
@@ -0,0 +1,27 @@
+# Custom Device Error Codes
+
+Historically custom devices have used error codes within the user-defined ranges.
+This has led to unintentional error code conflicts between different custom devices.
+
+To mitigate this, VeriStand custom devices now have a dedicated error code range: `732000` to `732999`, and `-732000` to `-732999`.
+Individual custom devices may be assigned a portion of this range; this document is the source of truth for these allocations.
+
+## Requesting an Error Code Range
+
+A new error code range can be requested by creating a pull request editing this document.
+
+In general, custom devices should reserve either `25` or `50` error codes (per positive/negative range).
```

Comment:
I made up these numbers - should we use different ones?

Reply:
To balance the need to constantly modify this file with the burden of requesting additional reserved error code ranges, I suggest that we reserve codes in blocks of 10 unless the custom device uses more custom error codes. The error codes do not have to be continuous if we use this document to track CD code ranges.

Reply:
We reserved a chunk of 1000 codes, so I would rather get contiguous ranges and slightly over-reserve. How about we reserve chunks of 25 or 50, but we leave 25 in between ranges in case they need to expand. @buckd have a tiebreaker thought?

Reply:
I think reserving 25 and a 25 code gap seems sufficient. I don't think many custom devices would use more than 25, but it can't hurt, given the number of custom devices we have and the number of codes reserved. I would prefer to have continuous codes per custom device.

Reply:
I agree with @Karl-G1 and @buckd. I'd prefer to have the error codes be contiguous if possible, and allocating 25 error codes per custom device but aligned to 50-width boundaries seems like the best option to do it while minimizing over-allocating.